src/graphic/text/bidi.cc

```diff
@@ -725,8 +742,16 @@ UI::FontSet const* find_fontset(const char* word, const UI::FontSets& fontsets)
 	UI::FontSets::Selector selector;
 	if (has_script_character(word, UI::FontSets::Selector::kArabic)) {
 		selector = UI::FontSets::Selector::kArabic;
+	} else if (strcmp(word, "简体中文") == 0 || strcmp(word, "繁體中文") == 0) {
+		selector = UI::FontSets::Selector::kCJK;
+	} else if (strcmp(word, "日本語") == 0) {
+		selector = UI::FontSets::Selector::kJapanese;
+	} else if (get_locale().rfind("ja", 0) == 0 && has_script_character(word, UI::FontSets::Selector::kJapanese)) {
```

Comment:
`has_script_character` is expensive, so nest this differently so we won't have to run it twice. Something like this:

```
if (has_script_character(word, UI::FontSets::Selector::kCJK)) {
    if (get_locale().rfind("ja", 0) == 0) // Japanese
    else // CJK
} else if (has_script_character(word, UI::FontSets::Selector::kJapanese)) ...
```

Reply:
P.S. or store the result of  `(has_script_character(word, UI::FontSets::Selector::kJapanese))` in a variable to avoid calling it twice.